Main Model/misc-observer-procedures.nls

```diff
@@ -1,55 +1,107 @@
-  report (1 + (random-float 1 - 0.5) / 50)
-end
 
 ;observer procedure, initializes a single consumer link
 to init-consumer-link
   hide-link
 end
 
-;observer procedure, sets the size of firms based on the number of employees they have
-;to set-firm-size
-;  set size (count employment-link-neighbors / 10) ; sqrt (count employment-link-neighbors / 3)
-;end
-
-;observer procedure, sets the size of households based on liquidity
-;to set-house-size
-;  set size  (sqrt (abs liquidity) / 12)
-;end
-
 ;observer procedure, initializes a single employment link
 to init-employment-link
   set color blue
   hide-link
 end
 
+;observer procedure, initializes a single equity link
+to init-equity-link
+  hide-link
+end
+
 ;observer procedure, initializes a single framework agreement
 to init-framework-agreement [input-type]
-  set input-firm-type input-type
   set demand-not-satisfied 0
   hide-link
 end
 
-;obsersver procedure, reports whether it is the beginning of a month
-to-report beginning-of-month?  
-  report ticks mod month-length = 0
+;observer procedure, recalculates the current index
+to recalculate-index-pringle
+  set pringle-index-value sum [equity] of equity-links         ;fixed firm implementation fixes index to be equal to firm number
+  ;NEED UPDATE
+end
+
+;observer procedure, recalculates the current index
+to recalculate-index-coats
+  set coats-index-value sum [price] of firms 
 end
 
-;observer procedure, reports whether it is the end of a month
-to-report end-of-month?  
-  report ticks mod month-length = (month-length - 1)
+;observer procedure, recalculates the current index
+to recalculate-index-ussher
+  ;IF WE DO ROLLING INVENTORY, ADD IN LIST OF 24 INVENTORIES AND AVERAGE OVER WHOLE 
+  set ussher-index-value sum [inventory * price] of firms
 end
 
+;observer procedure, recalculates the current index
+to recalculate-index-potvin
+  set potvin-index-value 1
+  ;implementation TBD
+end
+
+to set-colors
+  ask CONSUMER-GOOD-FIRMS [
+    set color lime + 3
+  ]
+  ask PRIMARY-GOOD-FIRMS [
+    set color brown
+  ]
+  
+  ask INTERMEDIATE-GOOD-FIRMS [
+    set color gray
+  ]
+  ask households [
+    ifelse unemployed? [
+      set color red
+    ] [
+      set color blue
+    ]
+  ]
+end
 
+;observer procedure, recalculates rolling average of the sum of past two years inventory
+to update-rolling-mean-inventory
+  ifelse month < 24 [
+    set LAST-TWO-YEARS-INVENTORY lput (sum [inventory] of firms) LAST-TWO-YEARS-INVENTORY 
+  ] [
+    set LAST-TWO-YEARS-INVENTORY replace-item (month mod 24) LAST-TWO-YEARS-INVENTORY (sum [inventory] of firms)
+  ]
+end
+
+;observer procedure to generate the correct location from the firms' liquidity
+to set-firm-xcor
+  ask firms [
+    let firm-liq-dev (max [liquidity] of firms - min [liquidity] of firms) / (max-pxcor - min-pxcor)   ;the difference in liquidity between patches 
+    let firm-num-dev liquidity - min [liquidity] of firms    ;the current households liquidity above the minimum
```

Comment:
Lets get rid of firm-num-dev and just set xcor to be `min-pxcor liquidity / firm-liq-dev.` If a firm isn't actually close to bankruptcy, I don't want it to look like they are. So, that also means that firm-liq-dev should just be max liquidity of firms divided by the `world-width` (which by the way you can use instead of amx-pxcor - min-pxcor)